pkg/validate/strategies.go

```diff
@@ -15,6 +17,10 @@ import (
 	"sigs.k8s.io/controller-runtime/pkg/client"
 )
 
+var (
+	systemReservedParametersRegexp = regexp.MustCompile(`BUILDER_IMAGE|DOCKERFILE|CONTEXT_DIR|(shp-.*)`)
```

Comment:
This shouldn't be duplicated, just share and export a method.

Reply:
n/a anymore.